### ISSUE TITLE
UnmarshalException doing any request

### ISSUE THREAD

**Issue by mmarin-3m19** [postaddictme/instagram-java-scraper#97]
Suddenly today when I try to do any request I got this exception:

```
Exception in thread "main" java.lang.IllegalArgumentException: javax.xml.bind.UnmarshalException
 - with linked exception:
[Exception [EclipseLink-25008] (Eclipse Persistence Services - 2.7.0.v20170811-d680af5): org.eclipse.persistence.exceptions.XMLMarshalException
Exception Description: A descriptor with default root element logging_page_id was not found in the project]
	at me.postaddict.instagram.scraper.mapper.ModelMapper.mapObject(ModelMapper.java:156)
	at me.postaddict.instagram.scraper.mapper.ModelMapper.mapAccount(ModelMapper.java:39)
	at me.postaddict.instagram.scraper.request.GetAccountRequest.mapResponse(GetAccountRequest.java:39)
	at me.postaddict.instagram.scraper.request.GetAccountRequest.mapResponse(GetAccountRequest.java:13)
	at me.postaddict.instagram.scraper.request.PaginatedRequest.requestInstagramResult(PaginatedRequest.java:39)
	at me.postaddict.instagram.scraper.Instagram.getAccount(Instagram.java:95)
	at me.postaddict.instagram.scraper.Instagram.getAccountByUsername(Instagram.java:85)
	at com.ca.instagram_connector_test.App.instagramScrapper(App.java:66)
	at com.ca.instagram_connector_test.App.main(App.java:29)
Caused by: javax.xml.bind.UnmarshalException
 - with linked exception:
[Exception [EclipseLink-25008] (Eclipse Persistence Services - 2.7.0.v20170811-d680af5): org.eclipse.persistence.exceptions.XMLMarshalException
Exception Description: A descriptor with default root element logging_page_id was not found in the project]
	at org.eclipse.persistence.jaxb.JAXBUnmarshaller.handleXMLMarshalException(JAXBUnmarshaller.java:1110)
	at org.eclipse.persistence.jaxb.JAXBUnmarshaller.unmarshal(JAXBUnmarshaller.java:172)
	at me.postaddict.instagram.scraper.mapper.ModelMapper.mapObject(ModelMapper.java:154)
	... 8 more
Caused by: Exception [EclipseLink-25008] (Eclipse Persistence Services - 2.7.0.v20170811-d680af5): org.eclipse.persistence.exceptions.XMLMarshalException
Exception Description: A descriptor with default root element logging_page_id was not found in the project
	at org.eclipse.persistence.exceptions.XMLMarshalException.noDescriptorWithMatchingRootElement(XMLMarshalException.java:162)
	at org.eclipse.persistence.internal.oxm.record.SAXUnmarshallerHandler.startElement(SAXUnmarshallerHandler.java:309)
	at org.eclipse.persistence.internal.oxm.record.json.JsonStructureReader.parsePair(JsonStructureReader.java:455)
	at org.eclipse.persistence.internal.oxm.record.json.JsonStructureReader.parseRoot(JsonStructureReader.java:181)
	at org.eclipse.persistence.internal.oxm.record.json.JsonStructureReader.parse(JsonStructureReader.java:139)
	at org.eclipse.persistence.internal.oxm.record.SAXUnmarshaller.unmarshal(SAXUnmarshaller.java:938)
	at org.eclipse.persistence.internal.oxm.record.SAXUnmarshaller.unmarshal(SAXUnmarshaller.java:414)
	at org.eclipse.persistence.internal.oxm.record.SAXUnmarshaller.unmarshal(SAXUnmarshaller.java:390)
	at org.eclipse.persistence.internal.oxm.XMLUnmarshaller.unmarshal(XMLUnmarshaller.java:394)
	at org.eclipse.persistence.jaxb.JAXBUnmarshaller.unmarshal(JAXBUnmarshaller.java:156)
	... 9 more
```

Checking your code, the exception comes from the requestInstagramResult method in the class PaginatedRequest. 

Did you have any idea of what is happening?

Regards,
Thanks a lot.

**Comment by igor-suhorukov** [comment]
Hi @mmarin-3m19 

Could you please use logger and error handler on httpClient and provide more details?
```
HttpLoggingInterceptor loggingInterceptor = new HttpLoggingInterceptor();
loggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);

OkHttpClient httpClient = new OkHttpClient.Builder()
        .addNetworkInterceptor(loggingInterceptor)
        .addInterceptor(new ErrorInterceptor())
        .cookieJar(new DefaultCookieJar(new CookieHashSet()))
        .build();
        client = new Instagram(httpClient);
client.basePage();
```

**Comment by igor-suhorukov** [comment]
@mmarin-3m19 looks like classpath issue

**Comment by mmarin-3m19** [comment]
I only added this dependency to the pom

```
<!-- https://mvnrepository.com/artifact/com.github.igor-suhorukov/instagramscraper -->
<dependency>
    <groupId>com.github.igor-suhorukov</groupId>
    <artifactId>instagramscraper</artifactId>
    <version>2.0</version>
</dependency>
```

I need to add another dependency?

And it was working fine but suddenly without any change I got that error.

**Comment by kjh487** [comment]
same error too...

**Comment by igor-suhorukov** [comment]
com.github.igor-suhorukov:instagramscraper:2.0 is correct dependency. Could you please provide more information about JRE/JDK version? 

> same error too...
 java.lang.NoClassDefFoundError: okhttp3/internal/Platform ?

**Comment by farseev** [comment]
**Same here...
Pls fix soon if possible - we have a research project show which involves it. Need to show it asap.**

- java-8-openjdk-amd64
- LSB Version:    core-9.20160110ubuntu0.2-amd64:core-9.20160110ubuntu0.2-noarch:printing-9.20160110ubuntu0.2-amd64:printing-9.20160110ubuntu0.2-noarch:security-9.20160110ubuntu0.2-amd64:security-9.20160110ubuntu0.2-noarch
- Distributor ID: Ubuntu
- Description:    Ubuntu 16.04.2 LTS
- Release:        16.04
- Codename:       xenial

**Exception details**
java.lang.IllegalArgumentException: javax.xml.bind.UnmarshalException
 - with linked exception:
[Exception [EclipseLink-25008] (Eclipse Persistence Services - 2.7.0.v20170811-d680af5): org.eclipse.persistence.exceptions.XMLMarshalException
Exception Description: A descriptor with default root element logging_page_id was not found in the project]
	at me.postaddict.instagram.scraper.mapper.ModelMapper.mapObject(ModelMapper.java:161) ~[followers-crawler-node-1391.jar:?]
	at me.postaddict.instagram.scraper.mapper.ModelMapper.mapAccount(ModelMapper.java:40) ~[followers-crawler-node-1391.jar:?]
	at me.postaddict.instagram.scraper.request.GetAccountRequest.mapResponse(GetAccountRequest.java:39) ~[followers-crawler-node-1391.jar:?]
	at me.postaddict.instagram.scraper.request.GetAccountRequest.mapResponse(GetAccountRequest.java:13) ~[followers-crawler-node-1391.jar:?]
	at me.postaddict.instagram.scraper.request.PaginatedRequest.requestInstagramResult(PaginatedRequest.java:39) ~[followers-crawler-node-1391.jar:?]
	at me.postaddict.instagram.scraper.Instagram.getAccount(Instagram.java:95) ~[followers-crawler-node-1391.jar:?]
	at me.postaddict.instagram.scraper.Instagram.getAccountByUsername(Instagram.java:85) ~[followers-crawler-node-1391.jar:?]
	at ai.somin.crawler.instagram.InstagramFollowersCrawler.crawl(InstagramFollowersCrawler.java:45) ~[followers-crawler-node-1391.jar:?]
	at ai.somin.crawler.followers.node.FollowersCrawlerNode.processMessage(FollowersCrawlerNode.kt:76) ~[followers-crawler-node-1391.jar:?]
	at ai.somin.crawler.followers.node.FollowersCrawlerNode.processMessage(FollowersCrawlerNode.kt:21) ~[followers-crawler-node-1391.jar:?]
	at net.bbridge.transport.messaging.NodeTransportConnection$CollectorCallbackImpl.onDataCollected(NodeTransportConnection.java:166) ~[followers-crawler-node-1391.jar:?]
	at net.bbridge.transport.collector.ForwardingMessageCollector.collect(ForwardingMessageCollector.java:18) ~[followers-crawler-node-1391.jar:?]
	at net.bbridge.transport.messaging.NodeTransportConnection$ConsumerImpl.lambda$handleDelivery$3(NodeTransportConnection.java:92) ~[followers-crawler-node-1391.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_111-internal]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_111-internal]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_111-internal]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_111-internal]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111-internal]
Caused by: javax.xml.bind.UnmarshalException
	at org.eclipse.persistence.jaxb.JAXBUnmarshaller.handleXMLMarshalException(JAXBUnmarshaller.java:1110) ~[followers-crawler-node-1391.jar:?]
	at org.eclipse.persistence.jaxb.JAXBUnmarshaller.unmarshal(JAXBUnmarshaller.java:172) ~[followers-crawler-node-1391.jar:?]
	at me.postaddict.instagram.scraper.mapper.ModelMapper.mapObject(ModelMapper.java:159) ~[followers-crawler-node-1391.jar:?]
	... 17 more
Caused by: org.eclipse.persistence.exceptions.XMLMarshalException: 
Exception Description: A descriptor with default root element logging_page_id was not found in the project
	at org.eclipse.persistence.exceptions.XMLMarshalException.noDescriptorWithMatchingRootElement(XMLMarshalException.java:162) ~[followers-crawler-node-1391.jar:?]
	at org.eclipse.persistence.internal.oxm.record.SAXUnmarshallerHandler.startElement(SAXUnmarshallerHandler.java:309) ~[followers-crawler-node-1391.jar:?]
	at org.eclipse.persistence.internal.oxm.record.json.JsonStructureReader.parsePair(JsonStructureReader.java:455) ~[followers-crawler-node-1391.jar:?]
	at org.eclipse.persistence.internal.oxm.record.json.JsonStructureReader.parseRoot(JsonStructureReader.java:181) ~[followers-crawler-node-1391.jar:?]
	at org.eclipse.persistence.internal.oxm.record.json.JsonStructureReader.parse(JsonStructureReader.java:139) ~[followers-crawler-node-1391.jar:?]
	at org.eclipse.persistence.internal.oxm.record.SAXUnmarshaller.unmarshal(SAXUnmarshaller.java:938) ~[followers-crawler-node-1391.jar:?]
	at org.eclipse.persistence.internal.oxm.record.SAXUnmarshaller.unmarshal(SAXUnmarshaller.java:414) ~[followers-crawler-node-1391.jar:?]
	at org.eclipse.persistence.internal.oxm.record.SAXUnmarshaller.unmarshal(SAXUnmarshaller.java:390) ~[followers-crawler-node-1391.jar:?]
	at org.eclipse.persistence.internal.oxm.XMLUnmarshaller.unmarshal(XMLUnmarshaller.java:394) ~[followers-crawler-node-1391.jar:?]
	at org.eclipse.persistence.jaxb.JAXBUnmarshaller.unmarshal(JAXBUnmarshaller.java:156) ~[followers-crawler-node-1391.jar:?]
	at me.postaddict.instagram.scraper.mapper.ModelMapper.mapObject(ModelMapper.java:159) ~[followers-crawler-node-1391.jar:?]
	... 17 more

**Comment by farseev** [comment]
@igor-suhorukov

**Comment by Mehrshadow** [comment]
looks like instagram has updated its api and several things, and this project needs to beed updated

**Comment by z0mb1ek** [comment]
I use last commit dependency, java 8, same error

**Comment by eos1d3** [comment]
Instagram changed the request response JSON around 18 hrs ago:

`https://www.instagram.com/" + username + "/?__a=1&max_id=" + maxID`

The whole JSON has major changes so all current version can't work any more.

And the `max_id` seems useless now.

**Comment by igor-suhorukov** [comment]
I try to fix new Endpoint url, but mapping also changed. Pull request for this issue are welcome

**Comment by Mehrshadow** [comment]
any good news?
our project is down, need fix asap :(
@igor-suhorukov

**Comment by farseev** [comment]
@eos1d3 did you find a way to fix your particular needed requests?

**Comment by eos1d3** [comment]
@farseev Yes I do. It is easy for me to fix. But I do not use this repository and I write the whole library myself. I am trying to help. But after importing this repository to Eclipse, I am getting hundreds of errors. 

How to start and test this repository?

**Comment by chikpro** [comment]
As i understand, we need to improve **account-binding.json** for correct mapping.

**Comment by igor-suhorukov** [comment]
@farseev you ask me for ETA but you not mark with star this repository yet...

![screenshot from 2018-03-16 00-14-52](https://user-images.githubusercontent.com/10332206/37493898-a30edad6-28b7-11e8-9599-4c878ac9ee08.png)

**Comment by igor-suhorukov** [comment]
@eos1d3 you need [lombok plugin for eclipse](https://projectlombok.org/setup/eclipse)

**Comment by igor-suhorukov** [comment]
Hi chaps! I change endpoint, change API and fix mapping for medias mapping. But part of account mapping is not still ready (medias preview section). You can build library from master and try to fix remaining part of mapping in account-binding.json for method me.postaddict.instagram.scraper.StatelessInsta#getAccountByUsername

**Comment by igor-suhorukov** [comment]
@Mehrshadow please check and change your code to follow new API

**Comment by farseev** [comment]
@igor-suhorukov, sorry, as you can see from my profile, I use GitHub occasionally and still do not have widely adopted community habits. Fixed. now it is starred by me :)

Any forecast regarding the issue fix? Sorry for asking again, but we really need it and we are not sure if we will be efficient fixing it by ourselves since are not really familiar with the library and the Instagram page structures.

![image](https://user-images.githubusercontent.com/14229589/37500790-5bb09fc2-2906-11e8-8880-f332f149c5b7.png)

**Comment by igor-suhorukov** [comment]
Account mapping was fixed in master branch

**Comment by tolgaduzenli** [comment]
Hi Igor,

Thank you so much for quick fix. 

Are you planning to do new release?

**Comment by igor-suhorukov** [comment]
@tolgaduzenli I will release library to maven central

**Comment by igor-suhorukov** [comment]
@tolgaduzenli released as version 2.1. Waiting for repository sync
